engine/apps/alerts/tasks/notify_user.py

```diff
@@ -420,17 +486,137 @@ def perform_notification(log_record_pk):
+                    schedule_perform_notification_task,
+                    log_record_notification_triggered.pk,
+                    log_record_notification_triggered.alert_group_id,
+                )
+            )
+            notifications.delete()
+        else:
+            UserNotificationPolicyLogRecord.objects.bulk_create(log_records_to_create, batch_size=5000)
+
+            if not active_alert_group_ids or not is_notification_allowed:
+                task_logger.info(
+                    f"no alert groups to notify about or notification is not allowed for user "
+                    f"{user_notification_bundle.user_id}"
+                )
+                notifications.delete()
+            else:
+                notifications.filter(id__in=skip_notification_ids).delete()
+                bundle_uuid = uuid4()
+                notifications.update(bundle_uuid=bundle_uuid)
+                task_logger.info(
+                    f"perform bundled notification for alert groups with ids: {active_alert_group_ids}, "
+                    f"bundle_uuid: {bundle_uuid}"
+                )
+                if user_notification_bundle.notification_channel == UserNotificationPolicy.NotificationChannel.SMS:
+                    PhoneBackend.notify_by_sms_bundle_async(user_notification_bundle.user, bundle_uuid)
+
+        user_notification_bundle.notification_task_id = None
+        user_notification_bundle.last_notified_at = timezone.now()
+        user_notification_bundle.eta = None
+        user_notification_bundle.save(update_fields=["notification_task_id", "last_notified_at", "eta"])
+
+    for alert_group_id in active_alert_group_ids:
+        transaction.on_commit(partial(send_update_log_report_signal.apply_async, (None, alert_group_id)))
+
+    # update metric
+    transaction.on_commit(partial(update_metric_if_needed, user_notification_bundle.user, active_alert_group_ids))
 
-    task_logger.debug(f"LOG RECORD PK: {log_record_pk}")
-    task_logger.debug(f"LOG RECORD LAST: {UserNotificationPolicyLogRecord.objects.last()}")
+    task_logger.info(f"Finished send_bundled_notification for user_notification_bundle {user_notification_bundle_id}")
 
-    log_record = UserNotificationPolicyLogRecord.objects.get(pk=log_record_pk)
-    user_notification_action_triggered_signal.send(sender=send_user_notification_signal, log_record=log_record)
 
-    task_logger.debug("--- USER SIGNAL TOOK %s seconds ---" % (time.time() - start_time))
+# deprecated
+@shared_dedicated_queue_retry_task(
+    autoretry_for=(Exception,), retry_backoff=True, max_retries=0 if settings.DEBUG else None
+)
+def send_user_notification_signal(log_record_pk):
+    # Triggers user_notification_action_triggered_signal
+    # This signal is only connected to UserSlackRepresentative and triggers posting message to Slack about some
+    # FAILED notifications (see NotificationDeliveryStep and ERRORS_TO_SEND_IN_SLACK_CHANNEL).
+    # No need to call it here.
+    pass
```

Comment:
can this be removed?

Reply:
It can be removed after release, because we want to make sure there are no more such tasks in the celery queue. Otherwise, a celery exception may occur when the scheduled task is not found

Reply:
makes sense!

Reply:
I'll add a note to https://github.com/grafana/oncall/issues/4629 to remove it as part of that task